pgmpy/estimators/BaseConstraintEstimator.py

```
@@ -0,0 +1,240 @@
+    variables: list
+        The list of variables (columns) in the data.
+    """
+
+    def __init__(
+        self,
+        data: Optional[pd.DataFrame] = None,
+        independencies: Optional[Independencies] = None,
+        **kwargs,
+    ) -> None:
+        super(BaseConstraintEstimator, self).__init__(
+            data=data, independencies=independencies, **kwargs
+        )
+
+    def _get_potential_sepsets(
+        self,
+        u: Hashable,
+        v: Hashable,
+        temporal_ordering: Dict[Hashable, int],
+        graph: UndirectedGraph,
+        lim_neighbors: int,
+    ) -> Collection[Tuple]:
+        """
+        Generates potential separating sets for an edge (u, v) based on the
+        current graph and temporal ordering.
+
+        This method identifies neighbors of u and v (excluding each other) and
+        then generates combinations of these neighbors of a specified size
+        (`lim_neighbors`). Temporal ordering is used to filter out neighbors
+        that occur after both u and v.
+
+        Parameters
+        ----------
+        u: Hashable
+            The first node of the edge.
+        v: Hashable
+            The second node of the edge.
+        temporal_ordering: Dict[Hashable, int]
+            A dictionary mapping variables to their temporal order.
+        graph: UndirectedGraph
+            The current state of the undirected graph.
+        lim_neighbors: int
+            The size of the conditioning sets to be generated.
+
+        Returns
+        -------
+        Collection[Tuple]
+            An iterable of tuples, where each tuple is a potential separating set.
+        """
+        separating_set_u = set(graph.neighbors(u))
+        separating_set_v = set(graph.neighbors(v))
+        separating_set_u.discard(v)
+        separating_set_v.discard(u)
+
+        if temporal_ordering != {}:
+            max_order = min(
+                temporal_ordering.get(u, float("inf")),
```

We shouldn't add a default value here. Better to throw an error for more predictable behavior. I would instead add a validation step for the temporal ordering in the `estimate` methods. Could you create an issue for that?